### PR TITLE
docs: add how do I examples

### PR DESCRIPTION
**Pull request by jimtng**
* Add more examples in a similar structure to jython docs from https://openhab-scripters.github.io/openhab-helper-libraries/Guides/But%20How%20Do%20I.html
* Some minor changes to the examples on the main page
* Additions to the logging page
* Add safe navigation to the Ruby Basics page
* Fix an error in the cron documentation

### CODE REVIEW

**Comment by jimtng** [comment]
The "How Do I" page crams almost everything in it.

File: docs/examples/how_do_i.md [hunk 1]
```diff
@@ -0,0 +1,864 @@
...
+  received_command gOutdoorLights.members do
+  run do |event|
+    after 3.minutes, id: event.item do # Use the triggering item as the timer ID
+      event.item.off
+    end
+  end
+end
+```
+
+Furthermore, you can manipulate the managed timers using the built-in `timers[]` array.
```
Comment (boc-tothefuture) [inline docs/examples/how_do_i.md:661]:
```suggestion
Furthermore, you can manipulate the managed timers using the built-in `timers[]` hash.
```

File: docs/examples/how_do_i.md [hunk 2]
```diff
@@ -0,0 +1,864 @@
...
+
+```ruby
+# timers[] is a special array to access the timers created with an id
```
Comment (boc-tothefuture) [inline docs/examples/how_do_i.md:664]:
```suggestion
# timers[] is a special hash to access the timers created with an id
```

File: docs/usage/misc/logging.md [hunk 3]
```diff
@@ -29,3 +30,33 @@ rule 'foo' do
...
+The `logger` object has the following methods
+
+| Method            |
+| ----------------- |
+| info              |
+| warn              |
+| error             |
+| debug             |
+| trace             |
+| is_info_enabled?  |
```
Comment (boc-tothefuture) [inline docs/usage/misc/logging.md:45]: Unrelated to the review, but we should fix those names in another PR.  Should be info?, warn?, etc.

**Review by boc-tothefuture — CHANGES_REQUESTED** [review]

**Comment by boc-tothefuture** [comment]
Awesome work!  I saw one minor thing about array vs. hash and made a suggested change.  Another comment about logger names which I will open as a distinct issue.

**Comment by boc-tothefuture** [comment]
Once this merges I will update our thread on openhab community.

File: docs/examples/how_do_i.md (hunk 2 above)
Comment (jimtng) [reply docs/examples/how_do_i.md]: fixed

File: docs/examples/how_do_i.md (hunk 1 above)
Comment (jimtng) [reply docs/examples/how_do_i.md]: fixed

File: docs/usage/misc/logging.md (hunk 3 above)
Comment (jimtng) [reply docs/usage/misc/logging.md:45]: good catch!

**Comment by jimtng** [comment]
> Awesome work! I saw one minor thing about array vs. hash and made a suggested change. Another comment about logger names which I will open as a distinct issue.

Thanks for the review and spotting the errors. I've fixed them in the latest commit.

**Pull request merged**